internal/api/handler/create_resource_v1alpha1.go

```diff
@@ -0,0 +1,75 @@
+package handler
+
+import (
+	"context"
+
+	"connectrpc.com/connect"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	sigyaml "sigs.k8s.io/yaml"
+
+	"github.com/akuity/kargo/internal/yaml"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+type CreateResourceV1Alpha1Func func(
+	context.Context,
+	*connect.Request[svcv1alpha1.CreateResourceRequest],
+) (*connect.Response[svcv1alpha1.CreateResourceResponse], error)
+
+func CreateResourceV1Alpha1(kc client.Client) CreateResourceV1Alpha1Func {
+	validateProject := newProjectValidator(kc)
+	parseKubernetesManifest := yaml.NewKubernetesManifestParser(kc.Scheme())
+	return func(
+		ctx context.Context,
+		req *connect.Request[svcv1alpha1.CreateResourceRequest],
+	) (*connect.Response[svcv1alpha1.CreateResourceResponse], error) {
+		parsed, err := parseKubernetesManifest(req.Msg.GetManifest())
+		if err != nil {
+			return nil, connect.NewError(connect.CodeInvalidArgument, errors.Wrap(err, "parse manifest"))
+		}
+
+		var res []*svcv1alpha1.CreateResourceResult
+		for _, obj := range parsed {
+			if req.Msg.GetNamespace() != "" {
+				obj.SetNamespace(req.Msg.GetNamespace())
+			}
+			if err := validateProject(ctx, obj.GetNamespace()); err != nil {
```

Comment:
It seems conceivable to me -- or even likely -- that the manifests you receive could actually _contain_ the namespace/project resource. i.e. "Here's a chunk of YAML that is the project and everything in it."

`kubectl` (and `helm`) both have rules about the order in which resources are created.

I think it would be both onerous and unnecessary to replicate those rules in their entirety because I think there are only certain resources that we should support here. For instance -- no one should be using this to create `Service`, `Deployment`, or `Ingress` resources. (We're not trying to be a replacement for `kubectl`.)

So the order might be something like this:

1. Namespaces/projects
1. Secrets, Policies, Roles, RoleBindings
1. Stages
1. Promotions

Reply:
And then that order probably needs to be reversed for deletes.